atomix/cluster/src/main/java/io/atomix/raft/roles/AbstractRole.java

```diff
@@ -49,13 +52,25 @@ protected AbstractRole(final RaftContext raft) {
                 .build());
   }
 
-  /**
-   * Returns the Raft state represented by this state.
-   *
-   * @return The Raft state represented by this state.
-   */
   @Override
-  public abstract RaftServer.Role role();
+  public Either<RaftError, Boolean> shouldAcceptRequest(final RaftRequest request) {
```

Comment:
:upside_down_face: If we don't care about the value, the right member might as well be Void to really indicate you're not supposed to ascribe any meaning to it

Reply:
This hints we may want a `Try` subtype of `Either` :smile: